src/cloudWatchLogs/commands/copyLogResource.ts

```diff
@@ -7,31 +7,37 @@ import * as nls from 'vscode-nls'
 const localize = nls.loadMessageBundle()
 
 import * as vscode from 'vscode'
-import { parseCloudWatchLogsUri } from '../cloudWatchLogsUtils'
+import { isLogStreamUri, parseCloudWatchLogsUri } from '../cloudWatchLogsUtils'
 import { copyToClipboard } from '../../shared/utilities/messages'
 
-export async function copyLogStreamName(uri?: vscode.Uri): Promise<void> {
+export async function copyLogResource(uri?: vscode.Uri): Promise<void> {
     try {
+
         if (!uri) {
             // No URI = used command palette as entrypoint, attempt to get URI from active editor
             // should work correctly under any normal circumstances since the action only appears in command palette when the editor is a CloudWatch Logs editor
             uri = vscode.window.activeTextEditor?.document.uri
             if (!uri) {
-                throw new Error("Attempt to copy Uri that doesn't exist.")
+                throw new Error("Attempt to copy resource name for Uri that doesn't exist.")
             }
-        }
+        } 
         const parsedUri = parseCloudWatchLogsUri(uri)
-        const streamName = parsedUri.logGroupInfo.streamName
+        let resourceName
+        if(isLogStreamUri(uri)){
+            resourceName = parsedUri.logGroupInfo.streamName
+        } else {
+            resourceName = parsedUri.logGroupInfo.groupName
+        }
 
-        if (!streamName) {
-            throw new Error(`Unable to copy stream name for Uri that doesn\'t have stream. Attempted copy on ${uri}`)
+        if (!resourceName) {
+            throw new Error(`Unable to copy log resource name for Uri that doesn\'t have resource. Attempted copy on ${uri}`)
```

Comment:
while we're here let's make this message less verbose, it's almost hard to understand

```suggestion
            throw new Error(`Failed to copy log resource name for URI (missing stream or group?): ${uri}`)
```